integrations/access/accesslist/app.go

```diff
@@ -0,0 +1,251 @@
+/*
+Copyright 2023 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package accesslist
+
+import (
+	"context"
+	"time"
+
+	"github.com/gravitational/trace"
+	"github.com/jonboulle/clockwork"
+
+	alclient "github.com/gravitational/teleport/api/client/accesslist"
+	"github.com/gravitational/teleport/api/types"
+	"github.com/gravitational/teleport/api/types/accesslist"
+	"github.com/gravitational/teleport/api/utils/retryutils"
+	"github.com/gravitational/teleport/integrations/access/common"
+	"github.com/gravitational/teleport/integrations/access/common/recipient"
+	"github.com/gravitational/teleport/integrations/access/common/teleport"
+	"github.com/gravitational/teleport/integrations/lib"
+	"github.com/gravitational/teleport/integrations/lib/logger"
+	pd "github.com/gravitational/teleport/integrations/lib/plugindata"
+	"github.com/gravitational/teleport/lib/services"
+	"github.com/gravitational/teleport/lib/utils"
+	"github.com/gravitational/teleport/lib/utils/interval"
+)
+
+const (
+	// oneWeek is the number of hours in a week.
+	oneWeek = 24 * time.Hour * 7
+)
+
+type App[M MessagingBot] struct {
+	pluginName  string
+	apiClient   teleport.Client
+	accessLists services.AccessListsGetter
+	pluginData  *pd.CompareAndSwap[pd.AccessListNotificationData]
+	bot         M
```

Comment:
Why do we need generics here? What's wrong with this?

```suggestion
	bot         MessagingBot
```

This feels a lot like the first example in the Go blog for [when you shouldn't use generics](https://go.dev/blog/when-generics#dont-replace-interface-types-with-type-parameters)


(My personal heuristic is "what would I have here done prior to Go having generics?" If the answer to that question is still reasonable today, then there's no need to use the generics)

Reply:
Agreed.